src/v/storage/parser.cc

```diff
@@ -79,18 +79,31 @@ static model::record_batch_header header_from_iobuf(iobuf b) {
 }
 
 static ss::future<result<iobuf>> verify_read_iobuf(
-  ss::input_stream<char>& in, size_t expected, ss::sstring msg) {
+  ss::input_stream<char>& in,
+  size_t expected,
+  ss::sstring msg,
+  bool recover = false) {
     return read_iobuf_exactly(in, expected)
-      .then([msg = std::move(msg), expected](iobuf b) {
+      .then([msg = std::move(msg), expected, recover](iobuf b) {
           if (likely(b.size_bytes() == expected)) {
               return ss::make_ready_future<result<iobuf>>(std::move(b));
           }
-          stlog.error(
-            "Cannot continue parsing. recived size:{} bytes, expected:{} "
-            "bytes. context:{}",
-            b.size_bytes(),
-            expected,
-            msg);
+          if (!recover) {
+              stlog.error(
+                "cannot continue parsing. recived size:{} bytes, expected:{} "
```

Comment:
nit: received, same below

Reply:
nice catch looks like this is preserving the same previous misspelling. maybe we can catch a lot of this with a bulk spelling check